src/Arcus.WebApi.Logging/ExceptionHandlingMiddleware.cs

```diff
@@ -63,30 +63,56 @@ public async Task Invoke(HttpContext context, ILoggerFactory loggerFactory)
+            catch (Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException exception)
 #endif
             {
                 // Catching the `BadHttpRequestException` and using the `.StatusCode` property allows us to interact with the built-in ASP.NET components.
                 // When the Kestrel maximum request body restriction is exceeded, for example, this kind of exception is thrown.
 
-                LogException(loggerFactory, ex);
-                context.Response.StatusCode = ex.StatusCode;
+                ILogger logger = CreateLogger(loggerFactory);
+                LogException(logger, exception);
+
+                WriteFailureToResponse(exception, context, (HttpStatusCode) exception.StatusCode);
             }
-            catch (Exception ex)
+            catch (Exception exception)
             {
-                LogException(loggerFactory, ex);
-                context.Response.StatusCode = (int) HttpStatusCode.InternalServerError;
+                ILogger logger = CreateLogger(loggerFactory);
+                LogException(logger, exception);
+
+                WriteFailureToResponse(exception, context, HttpStatusCode.InternalServerError);
             }
         }
 
-        private void LogException(ILoggerFactory loggerFactory, Exception ex)
+        private ILogger CreateLogger(ILoggerFactory loggerFactory)
         {
             string categoryName = _getLoggingCategory() ?? String.Empty;
-            var logger = loggerFactory.CreateLogger(categoryName) ?? NullLogger.Instance;
+            ILogger logger = loggerFactory.CreateLogger(categoryName) ?? NullLogger.Instance;
+
+            return logger;
+        }
+
+        /// <summary>
+        /// Logs the caught <paramref name="exception"/> before the response it sent back.
+        /// </summary>
+        /// <param name="logger">The instance to track the caught <paramref name="exception"/>.</param>
+        /// <param name="exception">The caught exception during the application pipeline.</param>
+        protected virtual void LogException(ILogger logger, Exception exception)
+        {
+            logger.LogCritical(exception, exception.Message);
+        }
 
-            logger.LogCritical(ex, ex.Message);
+        /// <summary>
+        /// Determine the HTTP status code based on the caught exception.
+        /// </summary>
+        /// <param name="exception">The caught exception during the application pipeline.</param>
+        /// <param name="context">The context instance for the current HTTP request.</param>
+        /// <param name="defaultFailureStatusCode">The default HTTP status code for the failure that was determined by the caught <paramref name="exception"/>.</param>
+        /// <returns>An HTTP status code that represents the <paramref name="exception"/>.</returns>
+        protected virtual void WriteFailureToResponse(Exception exception, HttpContext context, HttpStatusCode defaultFailureStatusCode)
```

Comment:
Just not sure if the order of parameters is ok.
`exception` and `defaultFailureStatusCode` are both parameters which just provide you information whereas the `context` parameter is the one that you manipulate.   Maybe `context` should be the last parameter ?

Reply:
Done in https://github.com/arcus-azure/arcus.webapi/pull/342/commits/e6605a2f99528abc9e4c181d8a5e5b5b0ffcbdf3

Reply:
I'm glad I'm not the only one who finds such small things important :D